ts/src/alpaca.ts

```diff
@@ -298,6 +306,299 @@ export default class alpaca extends Exchange {
+         * @see https://docs.alpaca.markets/reference/getaccount-1
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {int} the current integer timestamp in milliseconds from the exchange server
+         */
+        // get the account level information
+        const accountInfo = await this.fetchAccountInfo ();
+        // get the positions on the account
+        const positions = await this.fetchOpenPositions ();
+        // combine the two inputs
+        const combinedBalanceData: Dict = {
+            'accountInfo': accountInfo,
+            'positions': positions,
+        };
+        return this.parseBalance (combinedBalanceData);
+    }
+
+    parseBalance (response): Balances {
+        // loop through the positions and add them to the dict
+        const milliseconds = this.milliseconds ();
+        const result: Dict = {
+            'info': response,
+            'timestamp': milliseconds,
+            'datetime': this.iso8601 (milliseconds),
+        };
+        // parse the account cash
+        const accountInfo = this.safeValue (response, 'accountInfo');
+        let currencyId = this.safeString (accountInfo, 'currency');
+        let code = this.safeCurrencyCode (currencyId);
+        const accountCash = this.account ();
+        accountCash['total'] = this.safeString (accountInfo, 'cash');
+        accountCash['free'] = this.safeString (accountInfo, 'cash');
+        accountCash['used'] = '0';
+        result[code] = accountCash;
+        // parse the positions
+        const positions = this.safeValue (response, 'positions');
+        for (let i = 0; i < positions.length; i++) {
+            const entry = positions[i];
+            currencyId = this.safeString (entry, 'symbol');
+            code = this.safeCurrencyCode (currencyId);
+            const position = this.account ();
+            position['total'] = this.safeString (entry, 'qty');
+            position['free'] = '0';
+            position['used'] = this.safeString (entry, 'qty');
+            result[code] = position;
+        }
+        return this.safeBalance (result);
+    }
+
+    async fetchTickers (symbols: Strings = undefined, params = {}): Promise<Tickers> {
+        /**
+         * @method
+         * @name alpaca#fetchTickers
+         * @see https://docs.alpaca.markets/reference/cryptosnapshots-1
+         * @param {string[]} symbols unified symbols of the markets to fetch the ticker for
+         * @param {string} loc Crypto location ('us' only valid entry)
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} a dictionary of [ticker structures]{@link https://docs.ccxt.com/#/?id=ticker-structure}
+         */
+        await this.loadMarkets ();
+        const loc = this.safeString (params, 'loc', this.safeString (this.options, 'location'));
+        const parsedSymbols = symbols.join (',');
```

Comment:
you're assuming `symbols` are available but they are optional here, we might need to add a explicit check to require them `(if (symbols === undefined) throw ...`

Reply:
Nice catch. Updated to :

```
await this.loadMarkets ();
        const loc = this.safeString (params, 'loc', this.safeString (this.options, 'location'));
        let parsedSymbols = undefined;
        if (symbols !== undefined) {
            symbols = this.marketSymbols (symbols);
            parsedSymbols = symbols.join (',');
        }
        const request: Dict = {
            'symbols': parsedSymbols,
            'loc': loc,
        };
```